File: matlab/plotpsteady.m
function plotpsteady;
% PLOTPSTEADY  Plot P = P(W) in steady state with various assumptions
% about the overburden pressure Po and the sliding speed |v_b|.
% Generates figures for dampnotes.pdf.

p = params();
CC = p.c1 / (p.c2 * p.A);

h0 = 1000.0;             % m
Po = p.rhoi * p.g * h0;  % Pa; base overburden
v0   = 100.0 / p.spera;  % m/s; base sliding rate

% full range of W for vb=0 and F&C cases
W = 0.0:0.0005:1.2*p.Wr;
Pno  = psteady(p,Po,0.0,W) / 1e5;
% compare Flowers and Clarke function with W_crit = W_r
PFC  = Po * (W / p.Wr).^(7/2) / 1e5;

set(0,'defaultlinelinewidth',3.0)
set(0,'defaultaxesfontsize',16.0)

figure(1)
plot(W,Pno,'b',W,PFC,'k--')
hold on
sty = ['b'; 'g'; 'r'; 'c'];
aa = [0.1 1.0 10.0];
for j=1:3
  vb = aa(j) * v0;
  sbcube = p.c1 * vb / (p.c2 * p.A);
  Wc = sbcube * p.Wr - Po^3 * p.Y0;
  Wc = Wc / (sbcube + Po^3);
  Pc = psteady(p,Po,vb,Wc);
  fprintf('for vb = %8.3f m/a, Wc = %9.6f m and Pc = %10.3f Pa\n',...
          vb * p.spera, Wc, Pc)
  W = Wc:0.0005:1.2*p.Wr;
  P = psteady(p,Po,vb,W) / 1e5;
  plot(W,P,sty(j+1,:))
  plot(Wc,Pc,'k.','markersize',30)
end
hold off
xlabel('W  (m)'), ylabel('P  (bar)')
axis([0 max(W) 0 1.3*Po/1e5])

figure(2)
bb = [2.0 1.0 0.5 0.2];  % H = 2000, 1000, 500, 200 m
for j=1:4
  myPo = bb(j) * Po;
  sbcube = p.c1 * v0 / (p.c2 * p.A);
  Wc = sbcube * p.Wr - myPo^3 * p.Y0;
  Wc = Wc / (sbcube + myPo^3);
  Pc = psteady(p,myPo,v0,Wc);
  W = Wc:0.0005:1.2*p.Wr;
  P = psteady(p,myPo,v0,W) / 1e5;
  plot(W,P,sty(j,:))
  hold on
  plot(Wc,Pc,'k.','markersize',30)
end
hold off
xlabel('W  (m)'), ylabel('P  (bar)')
axis([0 max(W) 0 1.15*2*Po/1e5])

end

File: codes/plotpsteady.py
#!/usr/bin/env python

import numpy as np
import matplotlib.pyplot as plt
import sys
import argparse

spera = 31556926.0

A     = 3.1689e-24    # ice softness (Pa-3 s-1)
c1    = 0.500         # m-1
c2    = 0.040         # [pure]
g     = 9.81          # m s-2
rhoi  = 910.0         # kg m-3
Wr    = 1.0           # m

Y0    = 0.001         # regularization; m

def criticalW(Po,vb):
  sbcube = c1 * vb / (c2 * A)
  Wc = Wr * (sbcube / (sbcube + Po**3.0))
  return Wc

def psteady(W,Po,vb):
  '''Computes P(W) in steady state.'''
  if np.any(Po < 0.0):
    print 'psteady() requires nonnegative overburden pressure Po'
    sys.exit(1)
  if np.any(vb < 0.0):
    print 'psteady() requires nonnegative sliding speed vb'
    sys.exit(2)
  sbcube = c1 * vb / (c2 * A)
  frac = np.maximum(0.0, Wr - W) / (W + Y0)
  P = Po - (sbcube * frac)**(1.0/3.0)
  P[P < 0.0] = 0.0
  return P

W = np.linspace(0.0,1.2*Wr,501)

fig = plt.figure(1,figsize=(6.0,4.0))
H = 1000.0
Po = rhoi * g * H
vb = [0.0, 10.0, 100.0, 1000.0]
for j in range(4):
  P = psteady(W,Po,vb[j]/spera)
  Wc = criticalW(Po,vb[j]/spera)
  plt.plot(Wc,0.0,'ko',markersize=8.0,markeredgecolor='k',markerfacecolor='k')
  plt.hold(True)
  plt.plot(W[W>=0.95*Wc],P[W>=0.95*Wc]/1.0e5,'k',lw=2.0)
  plt.text(W[40 + 40*j],(P[40 + 40*j]/1.0e5) - 7.0 + 5.0 * j,r'$|\mathbf{v}_b| =$ %d m/a' % vb[j],rotation=8.0*j)
  if j == 1:
    plt.plot([0.0, W[1]],[0.0, P[1]/1.0e5],'k',lw=2.0)
plt.plot(W,Po * (W/Wr)**3.5 / 1.0e5,'k--',lw=2.5)
plt.hold(False)

#plt.gca().set_aspect('equal')
#plt.gca().autoscale(tight=True)
plt.gca().set_clip_on(False)
plt.xlabel('W  (m)')
plt.ylabel('P  (bar)')
plt.axis([-0.02, 1.19, -3.0, 110.0])
plt.tight_layout()

#plt.show()
fig.savefig('psteady-vb.pdf', bbox_inches='tight')
